hostfactory/host_provider/test/cluster_test.py

```diff
@@ -111,55 +121,120 @@ def run_test_machine_type_removed(request_count):
-        self.assertEqual(c, 30, f"Expected 30 got {c}")
-        
+            [{"node.vm_size": "A", "weight": 1, "template_id": "whatever", 'capacity-failure-backoff': 500}, 9, False]
+            ], expected_new_nodes_list=[], expected_result="NoAllocationSelected") 
+        
+        result = cluster.allocation_strategy(mymock, logger, "whatever", 17, 500, {"A": 1, "B": 1}, vm_dist)
+        self.assertEqual(result.status, "NoAllocationSelected")
+        self.assertEqual(result.node_list, [])
+        
+        # Test weighted distribution
+        vm_dist = {"A":17, "B":0, "C": 7}
+        expected_node_list = []
+        for vm, c in vm_dist.items():
+            expected_node_list.extend(set_node_list(vm, c, 1))
         mymock = MockNodeMgr([
-            [{"node.vm_size": "A", "weight": 1, "template_id": "whatever", 'capacity-failure-backoff': 500}, 1, False],
-            ], expect_new_nodes=1) 
-        c = cluster.allocation_strategy(mymock, "whatever", 1, 500, {"A": 1, "B": 1, "C": 1})
-        self.assertEqual(c, 1, f"Expected 1 got {c}")
-        # Need to change to take it account cores
+            [{"node.vm_size": "A", "weight": 1, "template_id": "whatever", 'capacity-failure-backoff': 500}, 17, False],
+            [{"node.vm_size": "C", "weight": 1, "template_id": "whatever", 'capacity-failure-backoff': 500}, 7, False],
+            [{"weight": 1, "template_id": "whatever", "capacity-failure-backoff": 500}, 11, False]],  expected_new_nodes_list=expected_node_list) 
+        result = cluster.allocation_strategy(mymock, logger, "whatever", 35, 500, {"A": 1, "B": 1, "C": 1}, vm_dist)
+        self.assertEqual(result.status, "success")
+        
+        # Test single core allocation.
+        vm_dist = {"A":1, "B":0, "C": 0}
+        vm_weights = {"A": 8, "B": 2, "C": 2}
+        expected_node_list = []
+        for vm, c in vm_dist.items():
+            expected_node_list.extend(set_node_list(vm, c, vm_weights[vm]))
         mymock = MockNodeMgr([
-            [{"node.vm_size": "A", "weight": 1, "template_id": "whatever", 'capacity-failure-backoff': 500}, 34, False],
-            [{"node.vm_size": "B", "weight": 1, "template_id": "whatever", 'capacity-failure-backoff': 500}, 33, False],
-            [{"node.vm_size": "C", "weight": 1, "template_id": "whatever", 'capacity-failure-backoff': 500}, 33, False]
-            ], expect_new_nodes=100) 
-        c = cluster.allocation_strategy(mymock, "whatever", 100, 500, {"A": 32, "B": 16, "C": 8})
-        self.assertEqual(c, 100, f"Expected 100 got {c}")
+            [{"node.vm_size": "A", "weight": 1, "template_id": "whatever", 'capacity-failure-backoff': 500}, 1, False]],  expected_new_nodes_list=expected_node_list)
+        result = cluster.allocation_strategy(mymock, logger, "whatever", 1, 500, {"A": 8, "B": 1, "C": 1}, vm_dist)
+        self.assertEqual(result.status, "success")
+        
         
+        # vm_dist = {"A":17, "B":0, "C": 7}
```

Comment:
You can just delete this commented out code, if you need it again it will be in git history.